test/functional/tests/fault_injection/test_fault_injection_with_mounted_core.py

```diff
@@ -0,0 +1,186 @@
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(512, Unit.MebiByte)])
+        core_dev = core_dev.partitions[0]
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+
+    with TestRun.step("Add core device with xfs filesystem and mount it."):
+        core = cache.add_core(core_dev)
+        core.create_filesystem(Filesystem.xfs)
+        core.mount(mount_point)
+
+    with TestRun.step("Try to remove core from cache."):
+        try:
+            cache.remove_core(1, 1)
+        except Exception:
+            TestRun.LOGGER.info("Can't remove core as expected.")
+        finally:
+            caches_count = len(casadm_parser.get_caches())
+            if caches_count != 1:
+                TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+
+    with TestRun.step("Try to stop OpenCAS."):
+        try:
+            cache.stop()
+        except Exception:
+            TestRun.LOGGER.info("Can't stop OpenCAS as expected.")
+        finally:
+            caches_count = len(casadm_parser.get_caches())
+            if caches_count != 1:
+                TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+
+    with TestRun.step("Unmount core device."):
+        core.unmount()
+
+    with TestRun.step("Stop all caches."):
+        casadm.stop_all_caches()
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_add_occupied_core(cache_mode):
+    """
+        title: Fault injection test to adding occupied core to the OpenCAS.
+        description: |
+          False positive test of the ability to add core to OpenCAS
+          while its occupied by the another OpenCAS instance.
+        pass_criteria:
+          - The same core device cannot be used twice in CAS.
+    """
+    with TestRun.step("Prepare two caches and one core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(2, Unit.GibiByte), Size(2, Unit.GibiByte)])
+        cache_dev1 = cache_dev.partitions[0]
+        cache_dev2 = cache_dev.partitions[1]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(1, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
+
+    with TestRun.step("Start first OpenCAS instance"):
+        cache1 = casadm.start_cache(cache_dev1, cache_mode, force=True)
+
+    with TestRun.step("Add core device to first OpenCAS instance."):
+        cache1.add_core(core_dev)
+
+    with TestRun.step("Start second OpenCAS instance"):
+        cache2 = casadm.start_cache(cache_dev2, cache_mode, force=True)
+
+    with TestRun.step("try add the same core device to second OpenCAS instance."):
+        try:
+            cache2.add_core(core_dev)
+        except Exception:
+            TestRun.LOGGER.info("Can't add this core as expected.")
```

Comment:
Isn't that expection what we expect in this test?

Reply:
Yes, but I changed this test like the previous one, I deleted try-except block and used framework's 'run_expect_fail' method.